fortanix-vme/fortanix-vme-abi/src/lib.rs

```diff
@@ -108,6 +112,32 @@ pub enum Response {
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub enum Error {
     ConnectionNotFound,
+    SystemError(i32),
+    Unknown,
+}
+
+#[cfg(feature="std")]
+impl From<io::Error> for Error {
+    fn from(error: io::Error) -> Error {
+        if let Some(errno) = error.raw_os_error() {
+            Error::SystemError(errno)
+        } else {
+            Error::Unknown
+        }
+    }
+}
+
+#[cfg(feature="std")]
+impl From<VsockError> for Error {
+    fn from(error: VsockError) -> Error {
+        match error {
+            VsockError::EntropyError        => Error::Unknown,
+            VsockError::SystemError(errno)  => Error::SystemError(errno),
+            VsockError::WrongAddressType    => Error::Unknown,
+            VsockError::ZeroDurationTimeout => Error::Unknown,
+            VsockError::ReservedPort        => Error::Unknown,
```

Comment:
This is not ideal. It's unlikely that a `VsockError` will ever be returned to an enclave, and even less that the enclave is able the handle it properly, but still we need to return something and `Unknown` may not be ideal.

Reply:
We currently only have three variants of the `Error` enum. Is there a reasoning/strategy behind this?

Reply:
For many errors, we simply want to forward the error to the enclave. When for example binding to a TCP port fails, the error can likely be translated to a `SystemError(errno)`, send back to the enclave, and there transformed again into a similar error.
Logical errors may arise in the runner that require their own error variant. For now there only is `ConnectionNotFound` that is returned when the enclave requests information about a non-existent connection.